chrysom/client.go

```diff
@@ -106,6 +107,10 @@ type ClientConfig struct {
 	// (Optional) If section is not provided with Listener, this
 	// feature will be disabled for the client.
 	Listen ListenerConfig
+
+	// HeaderConfig provides the placeholder names to be used in logger for traceID
+	// and spanID. (Optional) If HeaderConfig is not provided the defaults will be used.
+	HeaderConfig candlelight.HeaderConfig
```

Comment:
These values are only being used for logging - do logger keys need to be configurable? If we just want the trace information, constants such as `spanID` or `traceID` seem fine.  If this is meant to provide http header values, I think there is a better solution than cherry-picking and recreating header key-value pairs.

It seems like we're looking to include request headers as a part of the logging.  Passing a context with http values indicates to me that there is likely a logger in the context with http request headers already added as key-value pairs.  We can add a `GetLogger(ctx context.Context) log.Logger` function as part of the client configuration and then use that to get a logger instead of using the logger provided in the client.

Reply:
Using the GetLogger function provided by the client, I was not able able to add the GetLogger function as a part of the ClientConfig struct its test cases were failing because of function comparison.